runner/internal/timestats/timestats.go

```diff
@@ -7,73 +7,82 @@ import (
 )
 
 type TimeStats struct {
-	Min, Max, Avg, Median, StdDev time.Duration
-	Quartiles                     [4]time.Duration
-	Deciles                       [10]time.Duration
+	Min, Max, Mean, Median, StdDev time.Duration
+	Quartiles                      [4]time.Duration
+	Deciles                        [10]time.Duration
 }
 
-func Compute(times []time.Duration) (timeStats TimeStats) {
-	n := len(times)
-	if n == 0 {
-		return
+func Compute(times []time.Duration) TimeStats {
+	l := len(times)
+	if l == 0 {
+		return TimeStats{}
 	}
 
-	var sum, avg time.Duration
-	comparableDurations := make(comparableDurations, n)
-	for i, time := range times {
-		comparableDurations[i] = time
-		sum += time
-	}
+	// Measures computing functions works on sorted data.
+	// Sort once and compute upon the result.
+	sort.Sort(byFastest(times))
 
-	sort.Sort(comparableDurations)
-	avg = sum / time.Duration(n)
+	// Reused statistics measures.
+	sum := computeSum(times)
+	mean := computeMean(sum, l)
 
 	return TimeStats{
-		Min:       comparableDurations[0],
-		Max:       comparableDurations[len(comparableDurations)-1],
-		Avg:       avg,
-		Median:    calculateMedian(comparableDurations),
-		StdDev:    time.Duration(calculateStdDev(times, avg)),
-		Quartiles: calculateQuartiles(comparableDurations),
-		Deciles:   calculateDeciles(comparableDurations),
+		Min:       times[0],
+		Max:       times[len(times)-1],
+		Mean:      mean,
+		Median:    computeMedian(times),
+		StdDev:    computeStdDev(times, mean),
+		Quartiles: computeQuartiles(times),
+		Deciles:   computeDeciles(times),
+	}
+}
+
+func computeSum(values []time.Duration) time.Duration {
+	var sum time.Duration
+	for _, time := range values {
+		sum += time
 	}
+	return sum
+}
+
+func computeMean(sum time.Duration, length int) time.Duration {
+	return sum / time.Duration(length)
 }
 
-func calculateMedian(sorted []time.Duration) time.Duration {
-	n := len(sorted)
-	if n%2 != 0 {
-		return sorted[(n/2)-1]
+func computeMedian(sorted []time.Duration) time.Duration {
+	l := len(sorted)
+	if l%2 != 0 {
+		return sorted[l/2]
 	}
-	return (sorted[(n/2)-1] + sorted[(n/2)]) / 2
+	return computeMean(sorted[l/2-1]+sorted[l/2+1], 2)
 }
 
-func calculateStdDev(values []time.Duration, avg time.Duration) time.Duration {
-	n := len(values)
+func computeStdDev(values []time.Duration, mean time.Duration) time.Duration {
 	sum := time.Duration(0)
 	for _, v := range values {
-		dev := v - avg
+		dev := v - mean
 		sum += dev * dev
 	}
-	return time.Duration(math.Sqrt(float64(sum / time.Duration(n))))
+	return time.Duration(math.Sqrt(float64(sum / time.Duration(len(values)))))
 }
 
-func calculateDeciles(sorted []time.Duration) [10]time.Duration {
+func computeDeciles(sorted []time.Duration) [10]time.Duration {
 	const numDecile = 10
 	if len(sorted) < numDecile {
 		return [10]time.Duration{0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
```

Comment:
As discussed before, shouldn't we return `nil` here so that the consumer knows the data is not exploitable without having to read its values?

If not, let at least remove the unnecessary initialisation (idem L71)

```suggestion
		return [10]time.Duration{}
```